File: algo/src/main/java/com/hartwig/actin/algo/evaluation/general/IsLegallyInstitutionalized.java
package com.hartwig.actin.algo.evaluation.general;

import com.hartwig.actin.PatientRecord;
import com.hartwig.actin.algo.datamodel.Evaluation;
import com.hartwig.actin.algo.datamodel.EvaluationResult;
import com.hartwig.actin.algo.datamodel.ImmutableEvaluation;
import com.hartwig.actin.algo.evaluation.EvaluationFunction;

import org.jetbrains.annotations.NotNull;

public class IsLegallyInstitutionalized implements EvaluationFunction {

    IsLegallyInstitutionalized() {
    }

    @NotNull
    @Override
    public Evaluation evaluate(@NotNull PatientRecord record) {
        return ImmutableEvaluation.builder().result(EvaluationResult.NOT_EVALUATED)
                .addPassSpecificMessages("Currently assumed that patient is not legally institutionalized")
                .build();
    }
}

File: algo/src/main/kotlin/com/hartwig/actin/algo/evaluation/general/IsLegallyInstitutionalized.kt
package com.hartwig.actin.algo.evaluation.general

import com.hartwig.actin.PatientRecord
import com.hartwig.actin.algo.datamodel.Evaluation
import com.hartwig.actin.algo.evaluation.EvaluationFactory
import com.hartwig.actin.algo.evaluation.EvaluationFunction

class IsLegallyInstitutionalized : EvaluationFunction {
    
    override fun evaluate(record: PatientRecord): Evaluation {
        return EvaluationFactory.notEvaluated(
            "Currently assumed that patient is not legally institutionalized", "Assumed patient is not legally institutionalized"
        )
    }
}